index.js

```diff
@@ -135,51 +148,60 @@ function FSDuplexStream (path, opts) {
     this.setDefaultEncoding(opts.writeEncoding)
   }
 
-  this.path = path
-  this.bytesRead = 0
-  this.bytesWritten = 0
-
-  this._readEncoding = (opts.readEncoding || null)
+  Object.defineProperty(this, 'path', {
+    value: path,
+    enumerable: true
+  })
 
-  this._writeQueue = null
-  this._allIsRead = false
-  this._readTarget = 0
-  this._writeTarget = 0
+  this[AllIsRead] = false
+  this[BytesRead] = 0
+  this[BytesWritten] = 0
+  this[FinishCalled] = false
+  this[ReadEncoding] = (opts.readEncoding || null)
+  this[ReadTarget] = 0
+  this[WriteQueue] = null
+  this[WriteTarget] = 0
 
   fs.open(path, 'r+', onOpen.bind(this))
+}
 
-  var origEmit = this.emit
-  this.emit = function (ev) {
-    if (ev === 'finish') {
-      debug('intercepting finish event')
-
-      this.emit = origEmit
-      onFinish.call(this)
-
-      return
-    }
+FSDuplexStream.prototype = Object.create(Duplex.prototype)
```

Comment:
Hmm, I've never done that before, but maybe I should...

The builtin `util.inherits` doesn't do it:

```js
exports.inherits = function(ctor, superCtor) {

  if (ctor === undefined || ctor === null)
    throw new TypeError('The constructor to "inherits" must not be ' +
                        'null or undefined');

  if (superCtor === undefined || superCtor === null)
    throw new TypeError('The super constructor to "inherits" must not ' +
                        'be null or undefined');

  if (superCtor.prototype === undefined)
    throw new TypeError('The super constructor to "inherits" must ' +
                        'have a prototype');

  ctor.super_ = superCtor;
  Object.setPrototypeOf(ctor.prototype, superCtor.prototype);
};
```

But Babel's `inherits` does it:

```js
function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });

  if (superClass) {
    Object.setPrototypeOf
      ? Object.setPrototypeOf(subClass, superClass)
      : subClass.__proto__ = superClass;
  }
}
```